app/controllers/auctions_controller.rb

```diff
@@ -1,4 +1,7 @@
 class AuctionsController < ApplicationController
+  before_action :authenticate!, except: [:index]
+  before_action :authenticate_admin!, except: [:index]
```

Comment:
I'm not sure you want to do *BOTH* these validations.

I'd think you'd want to validate, for some actions, that the user is a user *OR* is an admin, and for others that the user is an admin. I'm not 100% sure the code as you've written it does what you want. Have you tried out all the possibilities?